e2e-tests/src/config.rs

```diff
@@ -130,6 +204,27 @@ impl Config {
         )
         .await
     }
+
+    pub fn nodes_configs(&self) -> anyhow::Result<std::vec::IntoIter<NodeConfig>> {
+        self.validator_names()
+            .into_iter()
+            .zip(self.synthetic_network_urls())
+            .zip(self.nodes_ip_addresses()?)
+            .zip(get_validators_keys(self))
+            .map(
+                |(((node_name, synthetic_network_url), ip_address), account)| -> anyhow::Result<_> {
```

Comment:
Ugh, these nested pairs are somewhat ugly. Maybe you could at least hide this in a method that returns an iterator over a quadruple?

There is nothing like a `flat_zip`, right? I think that couldn't properly work in this typesystem...

Reply:
Since this crate already depended on `itertools` I just used its `izip` macro.